src/uu/tail/src/args.rs

```diff
@@ -242,6 +237,82 @@ impl Settings {
 
         Ok(settings)
     }
+
+    pub fn has_only_stdin(&self) -> bool {
+        self.inputs.iter().all(|input| input.is_stdin())
+    }
+
+    pub fn has_stdin(&self) -> bool {
+        self.inputs.iter().any(|input| input.is_stdin())
+    }
+
+    pub fn num_inputs(&self) -> usize {
+        self.inputs.len()
+    }
+
+    pub fn check_warnings(&self) -> CheckResult {
+        if self.retry {
+            if self.follow.is_none() {
+                show_warning!("--retry ignored; --retry is useful only when following");
+            } else if self.follow == Some(FollowMode::Descriptor) {
+                show_warning!("--retry only effective for the initial open");
+            }
+        }
+
+        let mut no_pid_support: Option<CheckResult> = None;
+        if self.pid != 0 {
+            if self.follow.is_none() {
+                show_warning!("PID ignored; --pid=PID is useful only when following");
+            } else if !platform::supports_pid_checks(self.pid) {
+                show_warning!("--pid=PID is not supported on this system");
+                no_pid_support = Some(CheckResult::NoPidSupport);
+            }
```

Comment:
I think this warning and the corresponding `settings.pid = 0` should stay in the argument parsing, because then there is a single point where `pid` is set, which I find more important than grouping all the warnings. Simplifies this function because it's sole purpose is now just printing warnings and it doesn't have to return anything.

Reply:
>  but the order of the warnings would be mixed up then

In light of https://github.com/uutils/coreutils/issues/4000, I'm not super concerned about the warning order.

If you do want to keep the order, I would keep it simple and just change the `pid` in place.

Reply:
#4000 is a cool issue. But keeping the order the way gnu's tail is producing them isn't a big task either. I'll try to keep the pid change in settings then.